internal/store/assessment.go

```diff
@@ -15,8 +15,9 @@ import (
 type Assessment interface {
 	List(ctx context.Context, filter *AssessmentQueryFilter) (model.AssessmentList, error)
 	Get(ctx context.Context, id uuid.UUID) (*model.Assessment, error)
-	Create(ctx context.Context, assessment model.Assessment, inventory api.Inventory) (*model.Assessment, error)
-	Update(ctx context.Context, assessmentID uuid.UUID, name *string, inventory *api.Inventory) (*model.Assessment, error)
+	Create(ctx context.Context, assessment model.Assessment, inventory *api.Inventory) (*model.Assessment, error)
+	UpdateName(ctx context.Context, assessmentID uuid.UUID, name string) (*model.Assessment, error)
+	AddSnapshot(ctx context.Context, assessmentID uuid.UUID, inventory *api.Inventory) error
```

Comment:
this method I would move it to Snapshot store.
Why not having a Snapshot model passed to this function to avoid that logic with the state here?